juju/juju
# Make mongo memory profile editable.

This makes the controller config value mongo-memory-profile editable.

It also updates the agentconfigupdater worker to notice when this is updated and restart the agents.

## QA steps
 * bootstrap a 2.5.2 controller, and enable-ha
 * then upgarde to the new agent
```
juju ssh -m controller 1
sudo grep mongo /var/lib/juju/agents/mach*/agent.conf
```
should see
```
mongoversion: 3.6.3/wiredTiger
mongomemoryprofile: default
```

* bootstrap new controller with this branch
```
juju enable-ha
# wait
juju ssh -m controller 1
sudo grep mongo /var/lib/juju/agents/mach*/agent.conf
```
should see
```
mongoversion: 3.6.3/wiredTiger
mongomemoryprofile: default
```
Then ...
```
juju controller-config mongo-memory-profile=low
juju ssh -m controller 1
sudo grep mongo /var/lib/juju/agents/mach*/agent.conf
```
should see
```
mongoversion: 3.6.3/wiredTiger
mongomemoryprofile: low
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1816325
https://bugs.launchpad.net/juju/+bug/1816329
